code/core/src/types/modules/api-stories.ts

```diff
@@ -21,6 +21,7 @@ export interface API_GroupEntry extends API_BaseEntry {
   type: 'group';
   parent?: StoryId;
   children: StoryId[];
+  tags: Tag[];
```

Comment:
It would make the TS API a bit cleaner indeed. But I doubt it would make sense as there's virtually no way the intersection of _all_ stories' tags makes sense to compute. It would likely be a waste of CPU resources to compute.

Reply:
As far as I understand here, you are propagating N-2 levels up the tree. The roots add a single extra level, which means one extra intersection per root, which means almost no computation compared to this change. For the benefit of a cleaner, more consistent API.

Reply:
You're right that it doesn't cost that much and would simplify API for all users. Let me give it a go and see where it takes me!

I wonder if it would make sense to just document it as an empty array for 'root' entries, as that's what we expect would happen?

Reply:
I also realise that I do indeed go down N-2, which was not intended. I wanted tag intersections computed as an intersection of direct children for obvious performance reasons.

I'll rework the code to do all tag intersection in a separate pass, starting from leaves and going back up the tree.